UnitsNet.Tests/CustomCode/QuantityInfos/QuantityInfoExtensionsTest.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed under MIT No Attribution, see LICENSE file at the root.
+// Copyright 2013 Andreas Gullberg Larsen (andreas.larsen84@gmail.com). Maintained at https://github.com/angularsen/UnitsNet.
+
+using System;
+using UnitsNet.Units;
+using Xunit;
+
+namespace UnitsNet.Tests.QuantityInfos;
+
+public class QuantityInfoExtensionsTest
+{
+    [Fact]
+    public void GetDefaultUnit_ThrowsException_WhenUnitSystemIsNull()
+    {
+        Assert.Throws<ArgumentNullException>(() => Length.Info.GetDefaultUnit(null!));
+    }
+
+    [Fact]
+    public void GetDefaultUnit_ThrowsArgumentException_WhenNoUnitsFoundForUnitSystem()
+    {
+        // simulating a combination of units that won't match anything
+        var unsupportedUnitSystem = new UnitSystem(new BaseUnits(
+            (LengthUnit)(-1),
+            (MassUnit)(-1),
+            (DurationUnit)(-1),
+            (ElectricCurrentUnit)(-1),
+            (TemperatureUnit)(-1),
+            (AmountOfSubstanceUnit)(-1),
+            (LuminousIntensityUnit)(-1)));
+
+        Assert.Throws<ArgumentException>(() => Length.Info.GetDefaultUnit(unsupportedUnitSystem));
+    }
+
+    [Fact]
+    public void GetDefaultUnit_ReturnsCorrectUnit_ForSpecificUnitSystem()
+    {
+        LengthUnit defaultUnit = Length.Info.GetDefaultUnit(UnitSystem.SI);
+
+        Assert.Equal(LengthUnit.Meter, defaultUnit);
+    }
+
+    [Fact]
+    public void GetDefaultUnit_ReturnsFirstUnit_WhenMultipleUnitsExistForUnitSystem()
+    {
+        // Arrange
+        var unitSystem = new UnitSystem(new BaseUnits(
+            LengthUnit.Decimeter,
+            (MassUnit)(-1),
+            (DurationUnit)(-1),
+            (ElectricCurrentUnit)(-1),
+            (TemperatureUnit)(-1),
+            (AmountOfSubstanceUnit)(-1),
+            (LuminousIntensityUnit)(-1)));
+
+        // Act
+        VolumeUnit defaultUnit = Volume.Info.GetDefaultUnit(unitSystem);
+
+        // Assert
+        Assert.Equal(VolumeUnit.CubicDecimeter, defaultUnit);
```

Comment:
I assume this isn't going to be random/flaky, but if it is prone to randomness or future changes we could make it slightly more robust by asserting it being one of all known possible units, e.g. CubicDecimeter and Liter. This does not help if adding new units in the future, of course.

Reply:
Yes I know, initially this was returning the `VolumeUnit.ImperialPint` 😆 and I had left a comment about it, but then I went on and fixed the json files... 
Still, I don't care- let it break, if somebody adds a `VolumeUnit` starting with "A, B, or C", that has the same `BaseUnits` give me call, I'll fix the test 😎

Reply:
This comment will come back to haunt you 😆

Reply:
Right, don't think I'll be loosing much sleep over it..